src/py21cmfast/src/filtering.c

```diff
@@ -17,64 +17,57 @@
 #include "indexing.h"
 #include "dft.h"
 
-void filter_box_annulus(fftwf_complex *box, int RES, float R_inner, float R_outer){
-    int n_x, n_z, n_y, dimension, midpoint;
-    float k_x, k_y, k_z, k_mag, kRinner, kRouter;
-    float f_inner, f_outer;
-
-    switch(RES) {
-        case 0:
-            dimension = user_params_global->DIM;
-            midpoint = MIDDLE;
-            break;
-        case 1:
-            dimension = user_params_global->HII_DIM;
-            midpoint = HII_MIDDLE;
-            break;
-        default:
-            LOG_ERROR("Resolution for filter functions must be 0(DIM) or 1(HII_DIM)");
-            Throw(ValueError);
-            break;
-    }
-    // loop through k-box
-
-#pragma omp parallel shared(box) private(n_x,n_y,n_z,k_x,k_y,k_z,k_mag,kRinner,kRouter,f_inner,f_outer) num_threads(user_params_global->N_THREADS)
-    {
-#pragma omp for
-        for (n_x=0; n_x<dimension; n_x++){
-            if (n_x>midpoint) {k_x =(n_x-dimension) * DELTA_K;}
-            else {k_x = n_x * DELTA_K;}
+double real_tophat_filter(double kR){
+    if (kR < 1e-4)
```

Comment:
To get slightly higher precision, you could use the taylor expansion around zero, even to second order (i.e. 1 - x^2/10). This gives accuracy to the full extent of double-precision at kR=1e-4